makefile.shade

```diff
@@ -1,45 +1,108 @@
-            {
-                sourceStream.CopyTo(stream);
-            }
-
-            return packagePart;
-        }
-    }
+use assembly='WindowsBase, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
+use namespace='System.IO.Packaging'
+use import="Environment"
+
+var VERSION='0.1'
+var FULL_VERSION='0.1'
+var AUTHORS='Microsoft Open Technologies, Inc.'
+
+use-standard-lifecycle
+k-standard-goals
+
+var Configuration2='${E("Configuration")}' 
+var ROOT = '${Directory.GetCurrentDirectory()}' 
+var BUILD_DIR2 = '${Path.Combine(ROOT, "build")}'
+
+#build-compile target='compile'
+
+#build-windows target='build-compile' if='IsWindows'
+#build-windows target='compile' if='!IsWindows && !IsBuildV2 && Directory.Exists("src")'
+    @{
+        var projectFiles = Files.Include("src/**/project.json")
+            .Exclude("src/Microsoft.AspNet.DataProtection.SystemWeb/project.json")
+            .ToList();
+
+        if (ShouldRunInParallel)
+        {
+            Parallel.ForEach(projectFiles, projectFile => DnuPack(projectFile, BUILD_DIR2, Configuration2));
+        }
+        else
+        {
+            projectFiles.ForEach(projectFile => DnuPack(projectFile, BUILD_DIR2, Configuration2));
+        }
+
+        foreach (var nupkg in Files.Include(Path.Combine(BUILD_DIR2, "*/*.nupkg")))
+        {
+            File.Copy(nupkg, Path.Combine(BUILD_DIR2, Path.GetFileName(nupkg)), true);
+        }
+    }
+
+#build-compile target='compile' if='!IsWindows && IsBuildV2'
+    @{
+        if (Directory.Exists("src"))
+        {
+            var projects = Files.Include("src/*")
+                .Exclude("src/Microsoft.AspNet.DataProtection.SystemWeb")
+                .ToList();
+            
+            DnuPack(string.Join(";", projects), BUILD_DIR2, Configuration2);
+
+            foreach (var nupkg in Files.Include(Path.Combine(BUILD_DIR2, "*/*.nupkg")))
+            {
+                File.Copy(nupkg, Path.Combine(BUILD_DIR2, Path.GetFileName(nupkg)), true);
+            }
+        }
+
+        if (Directory.Exists("test"))
+        {
+            DnuBuild("test/*", Configuration2);
+        }
+    }
+
+#nupkg-patch target='compile'
+    @{
+        if (Directory.Exists("artifacts"))
+        {
+            var packagePaths = Files.Include("artifacts/build/**/Microsoft.AspNet.DataProtection.SystemWeb.*.nupkg")
+                .Exclude("**/*.symbols.nupkg");
+            foreach (var packagePath in packagePaths)
+            {
+                using (var package = Package.Open(packagePath, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    CreatePartFromFile(
+                        package,
+                        @"src\Microsoft.AspNet.DataProtection.SystemWeb\web.config.transform",
+                        @"content\net451\web.config.transform");
+                }
+            }
+        }
+    }
+
+functions
+    @{
+        PackagePart CreatePartFromFile(

```

Comment:
We should be able to get rid of this and use the new feature that allows packing any file into the nupkg. See https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNet.Server.Kestrel/project.json#L62